source/sdk/math/Random.ooc

```diff
@@ -39,6 +40,17 @@ Random: class {
 	randFloat: static func -> Float {
 		return (rand() as Float) / INT_MAX
 	}
+	/**
+	   Box-Muller transform, http://en.wikipedia.org/wiki/Box-Muller_transform
+	   :return: a pseudo-random float number with a normal distribution
+	 */
+	generateGaussianFloat: static func (sigma := 1.0f, mu := 0.0f) -> Float {
+		result := 0.0f
+		u1 := randFloat()
+		u2 := randFloat()
```

Comment:
`randFloat()` can return 0. According to the [wikipedia page for Box-Muller](http://en.wikipedia.org/wiki/Box%E2%80%93Muller_transform), 
> The basic form as given by Box and Muller takes two samples from the uniform distribution on the interval (0, 1]

and `log(0)` is probably something you don't want to happen. I'm afraid you'll have to do 
```ooc
u1 := Float maximum(randFloat(), Float minimumValue)
```
to be on the safe side.